### PR TITLE
[15.0][MIG] product_variant_default_code: Migration to version 15.0

### PR DESCRIPTION
**Pull request by pilarvargas-tecnativa**
cc @Tecnativa TT36978

@stefan-tecnativa @chienandalu please review

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Check pre-commit

File: product_variant_default_code/views/product_view.xml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                          [('attribute_line_ids', '=', [])]}"
+                    groups="product_variant_default_code.group_product_default_code_manual_mask"
+                />
+            </field>
+            <xpath expr="//page/field[@name='attribute_line_ids']" position="before">
+                    <field
+                    name="variant_default_code_error"
+                    class="alert alert-danger alert-dismissable"
+                    role="alert"
+                    attrs="{'invisible':[('variant_default_code_error','=', False)]}"
```
Comment (chienandalu) [inline product_variant_default_code/views/product_view.xml:26]: This is useless to show if there aren't variants/attributes:

image

File: product_variant_default_code/README.rst [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+  - `JKT01-Bl-M` Jacket Black M
+  - `JKT01-Wh-L` Jacket White L
+  - `JKT01-Bl-L` Jacket Black L
+
+**Table of contents**
+
+.. contents::
+   :local:
+
+Usage
```
Comment (chienandalu) [inline product_variant_default_code/README.rst:74]: I'm missing a configuration section with the permissions needed and global configuration options.

**Review by chienandalu — COMMENTED** [review]
Tested. Some little remarks:

**Review by stefan-tecnativa — COMMENTED** [review]
Functionally tested. Just check the aspects commented by David.

File: product_variant_default_code/README.rst (hunk 2 above)
Comment (pilarvargas-tecnativa) [reply product_variant_default_code/README.rst:74]: Please check if the configuration file is correct to regenerate the readme.

File: product_variant_default_code/views/product_view.xml (hunk 1 above)
Comment (pilarvargas-tecnativa) [reply product_variant_default_code/views/product_view.xml:26]: solved

File: product_variant_default_code/README.rst (hunk 2 above)
Comment (stefan-tecnativa) [reply product_variant_default_code/README.rst:74]: It looks good to me.

File: product_variant_default_code/readme/CONFIGURATION.rst [hunk 3]
```diff
@@ -0,0 +1,15 @@
+To configure the default code of the product variant, it is necessary:
```
Comment (pedrobaeza) [inline product_variant_default_code/readme/CONFIGURATION.rst]: This should be called CONFIGURE.rst.

**Review by pedrobaeza — APPROVED** [review]

**Review by stefan-tecnativa — APPROVED** [review]

**Pull request merged**